src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+                        (this.input_el as HTMLInputElement)?.value
+                    )
+                );
+
+                this.setSuggestion(this.filtered_list, false);
+            } else {
+                this.filtered_list = [];
+
+                this.setSuggestion(this.filtered_list, false);
+            }
+        }
+        return;
+    }
+
+    setActive(resultLength: number) {
+        if (this.prevIndex >= 0 && this.prevIndex < resultLength) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
+
+            if (selectedItem) {
+                if (this.prevIndex != -1) {
+                    this.private_items[this.prevIndex].classList.remove(
+                        "sgn-active"
+                    );
+                }
+                selectedItem.classList.add("sgn-active");
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add("sgn-active");
+        }
+
+        this.prevIndex = this.index;
+    }
+
+    setInputValue() {
+        const selectedValue = this.result[this.index]?.name;
+        this.prevIndex = this.index;
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+        }
+    }
+    _highlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (input && name.toLowerCase().includes(input.toLowerCase())) {
+            const parts = name.split(new RegExp(`(${input})`, "gi"));
+
+            return parts
+                .map((part: any, index: number) => {
+                    if (index !== 0) {
+                        return `<span class="sgn-highlight">${part}</span>`;
+                    } else {
+                        return part;
+                    }
+                })
+                .join("");
+        }
+        return name;
+    }
+    _startsWithHighlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (input && name.toLowerCase().startsWith(input.toLowerCase())) {
+            const start = name.substring(0, input.length);
+            const rest = name.substring(input.length);
+            return `<span class="sgn-highlight" >${start}</span>${rest}`;
+        }
+        return name;
+    }
+    setSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more?: boolean
+    ) {
+        this.result = [...suggestions];
+        this.index = this.index;
+        this.input_temp = (this.input_el as HTMLInputElement)?.value;
+
+        this.input_temp ? (this.show_empty = false) : (this.show_empty = true);
+        this.has_more = has_more || false;
+        if (this._private_slot_el) {
```

Comment:
There is no need for the private_slot_el variable at all. You can revise this check with
`this.private_items?.length`